### PR TITLE
Feature/roles db/repush

### PR DESCRIPTION
**Pull request by andromedis**
#### Changes:

- Added migration and seed to create roles table in the database
- Edited migration and seed for profiles table, removed roles column and added role_id foreign key column
- Edited profile seed to have a moderator user for testing
- Reordered seed files and roles migration to ensure they run in the correct order
- Updated find functions in profileModel to join with roles table and return role_name of profiles
- Updated reference profile schema in documentation

#### Goal/Context: 

- Better database normalization
- Upcoming changes to how role permissions are handled to fix security vulnerability
- Upcoming addition of moderator role with case review permissions
- Needed more consistent way to reference roles

#### Fixes:

- Lack of database normalization around profile roles

## Type of change

- [X] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have removed unnecessary comments/console logs from my code
- [X] I have made corresponding changes to the documentation if necessary (optional)
- [X] My changes generate no new warnings
- [X] I have checked my code and corrected any misspellings
- [X] No duplicate code left within changed files
- [X] Size of pull request kept to a minimum
- [X] Pull request description clearly describes changes made & motivations for said changes

## Note: 

This is a duplicate of an earlier pull request written to remove commits from an earlier team that shouldn't be part of the PR. Suggested edits reviewers made on the earlier one were also made here.

### CODE REVIEW

File: api/profile/profileModel.js [hunk 1]
```diff
@@ -1,19 +1,28 @@
...
+    .where(filter);
 };
 
 const findById = async (user_id) => {
-  const user = await db('profiles').where({ user_id }).first().select('*');
+  const user = await db('profiles')
+    .leftJoin('roles', 'profiles.role_id', 'roles.role_id')
+    .where({ user_id })
+    .first();
   let book_marked_cases = await db('book_mark_cases').where({ user_id });
```
Comment (bukit3point0) [inline api/profile/profileModel.js:26]: Thanks for checking in on the `.select('*')`!

**Review by bukit3point0 — APPROVED** [review]
Thanks for reviewing the comments I had before! Looks good to me

**Review by kmcervan — COMMENTED** [review]
Looks good!

File: api/profile/profileModel.js (hunk 1 above)
Comment (elCarlosSantiago) [reply api/profile/profileModel.js:26]: +1 love the leftJoins!

File: data/seeds/002-profiles.js [hunk 2]
```diff
@@ -1,11 +1,11 @@
 // Only 8 Okta-approved profiles
-// ID 1 and 2 are admins, ID 3-8 are users
+// ID 1 and 2 are admins, ID 3 is moderator, 4-8 are users
 const profiles = [...new Array(9)].map((i, idx) => ({
```
Comment (elCarlosSantiago) [inline data/seeds/002-profiles.js:3]: Maybe we can make i into a more semantic variable for future devs. Functionally ok!

**Review by elCarlosSantiago — APPROVED** [review]
Great work Becca! Co-reviewed with Brett.

**Review by salvare736 — APPROVED** [review]
Clear and succinct changes. Good work!

**Pull request merged**